StaticAnalyzer/views/ios/code_analysis.py

```diff
@@ -2,62 +2,85 @@
-                            src, dirname, jfile.replace('+', 'x'))
-                        shutil.move(jfile_path, new_path)
-                        jfile_path = new_path
-                    dat = ''
-                    with io.open(jfile_path,
-                                 mode='r',
-                                 encoding='utf8',
-                                 errors='ignore') as flip:
-                        dat = flip.read()
-
-                    # Code Analysis
-                    relative_src_path = jfile_path.replace(src, '')
-                    code_rule_matcher(code_findings, [], dat,
-                                      relative_src_path, code_rules)
-                    # API Analysis
-                    api_rule_matcher(api_findings, [], dat,
-                                     relative_src_path, api_rules)
-
-                    # Extract URLs and Emails
-                    urls, urls_nf, emails_nf = url_n_email_extract(
-                        dat, relative_src_path)
-                    url_list.extend(urls)
-                    url_n_file.extend(urls_nf)
-                    email_n_file.extend(emails_nf)
+                    api_rules = ios_apis.CODE_APIS
+                    code_rules = objc_rules.OBJC_RULES
+                    source_types.add(_SourceType.objc)
+                elif jfile.endswith('.swift'):
+                    api_rules = ios_apis.CODE_APIS
+                    code_rules = swift_rules.SWIFT_RULES
+                    source_types.add(_SourceType.swift)
+                else:
+                    continue
+
+                jfile_path = os.path.join(src, dirname, jfile)
+                if '+' in jfile:
+                    new_path = os.path.join(
+                        src, dirname, jfile.replace('+', 'x'))
+                    shutil.move(jfile_path, new_path)
+                    jfile_path = new_path
+                dat = ''
+                with io.open(jfile_path,
+                             mode='r',
+                             encoding='utf8',
+                             errors='ignore') as flip:
+                    dat = flip.read()
+
+                # Code Analysis
+                relative_src_path = jfile_path.replace(src, '')
+                code_rule_matcher(code_findings, [], dat,
+                                  relative_src_path, code_rules)
+                # API Analysis
+                api_rule_matcher(api_findings, [], dat,
+                                 relative_src_path, api_rules)
+
+                # Extract URLs and Emails
+                urls, urls_nf, emails_nf = url_n_email_extract(
+                    dat, relative_src_path)
+                url_list.extend(urls)
+                url_n_file.extend(urls_nf)
+                email_n_file.extend(emails_nf)
+
+        if len(source_types) > 1:
+            source_type = _SourceType.swift_and_objc.value
+        else:
+            source_type = source_types.pop().value
```

Comment:
What if the zip package had no swift or objective c file (An edge case)?
 `source_types` will be an empty set. We need MobSF to handle exception gracefully and proceed with the rest of the checks even if one the process failed.